components/sup/src/manager/service/hooks.rs

```diff
@@ -66,33 +66,64 @@ pub trait Hook: fmt::Debug + Sized {
 
     fn file_name() -> &'static str;
 
+    fn deprecated_file_name() -> Option<&'static str> {
+        None
+    }
```

Comment:
This approach is good if we want to support underscores only for `file_updated`, `health_check` and `smoke_test`. But I wonder if it would be better UX to temporarily support (and deprecate) all hook names with underscores. That would lead to slightly simpler code because we could just calculate the deprecated name inline in `load` rather than add `deprecated_file_name` to the trait with something like:
```rust
        let deprecated_file_name = if Self::file_name().contains("_") {
            Some(Self::file_name().replace("_", "-"))
        } else {
            None
        };

```

Reply:
That looks simpler indeed!